x-pack/packages/kbn-elastic-assistant/impl/assistant/use_conversation/index.tsx

```diff
@@ -51,299 +48,264 @@ interface CreateConversationProps {
-  amendMessage: ({ conversationId, content }: AmendMessageProps) => void;
+  appendMessage: ({
+    conversationId,
+    message,
+  }: AppendMessageProps) => Promise<Message[] | undefined>;
+  amendMessage: ({ conversationId, content }: AmendMessageProps) => Promise<void>;
   appendReplacements: ({
     conversationId,
     replacements,
-  }: AppendReplacementsProps) => Record<string, string>;
-  clearConversation: (conversationId: string) => void;
-  createConversation: ({ conversationId, messages }: CreateConversationProps) => Conversation;
+  }: AppendReplacementsProps) => Promise<Record<string, string> | undefined>;
+  clearConversation: (conversationId: string) => Promise<void>;
+  getDefaultConversation: ({ conversationId, messages }: CreateConversationProps) => Conversation;
   deleteConversation: (conversationId: string) => void;
-  removeLastMessage: (conversationId: string) => Message[];
-  setApiConfig: ({ conversationId, apiConfig }: SetApiConfigProps) => void;
-  setConversation: ({ conversation }: SetConversationProps) => void;
+  removeLastMessage: (conversationId: string) => Promise<Message[] | undefined>;
+  setApiConfig: ({
+    conversation,
+    apiConfig,
+  }: SetApiConfigProps) => Promise<Conversation | IHttpFetchError<unknown>>;
+  createConversation: (conversation: Conversation) => Promise<Conversation | undefined>;
+  getConversation: (conversationId: string) => Promise<Conversation | undefined>;
 }
 
 export const useConversation = (): UseConversation => {
   const {
     allSystemPrompts,
     assistantTelemetry,
     knowledgeBase: { isEnabledKnowledgeBase, isEnabledRAGAlerts },
-    setConversations,
+    http,
   } = useAssistantContext();
 
+  const getConversation = useCallback(
+    async (conversationId: string) => {
+      const currentConversation = await getConversationById({ http, id: conversationId });
+      if (isHttpFetchError(currentConversation)) {
+        return;
+      }
+      return currentConversation;
+    },
+    [http]
+  );
+
   /**
    * Removes the last message of conversation[] for a given conversationId
    */
   const removeLastMessage = useCallback(
-    (conversationId: string) => {
+    async (conversationId: string) => {
       let messages: Message[] = [];
-      setConversations((prev: Record<string, Conversation>) => {
-        const prevConversation: Conversation | undefined = prev[conversationId];
-
-        if (prevConversation != null) {
-          messages = prevConversation.messages.slice(0, prevConversation.messages.length - 1);
-          const newConversation = {
-            ...prevConversation,
-            messages,
-          };
-          return {
-            ...prev,
-            [conversationId]: newConversation,
-          };
-        } else {
-          return prev;
-        }
-      });
+      const prevConversation = await getConversationById({ http, id: conversationId });
+      if (isHttpFetchError(prevConversation)) {
+        return;
```

Comment:
Consider displaying an error toast indicating the last message was not removed from the conversation